### PR TITLE
Hash for link

### PR DESCRIPTION
**Pull request by kota-yata**
#457 

Here is the test website => https://kota-yata.github.io/srihash-gh/

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -35,6 +35,44 @@ async function hashText(message) {
   return hash384;
 }
 
+function parseContentType(res) {
```
Comment (mozfreddyb) [inline app.js]: nit: The function should get the header value as a string, you should do the "res.headers.get()" in the caller of this function.

File: app.js [hunk 2]
```diff
@@ -35,6 +35,44 @@ async function hashText(message) {
...
+  const base64string = btoa(
+    String.fromCharCode(...new Uint8Array(hashBuffer))
+  );
+
+  return `sha384-${base64string}`;
+}
+
+async function displayResult(resultDiv, url, contentType, text) {
+  const integrity = await integrityMetadata(text);
+  const scriptEl = `<span style="color: #ffa07a">&lt;script src=</span><span style="color:#abe338">&quot;${encodeURI(
```
Comment (mozfreddyb) [inline app.js]: I think it's generally better to generate both result HTML strings within the if/else below. No need to create both and then just use one.

File: app.js [hunk 3]
```diff
@@ -49,18 +87,10 @@ async function formSubmit(event) {
 
     console.info("Response", response);
     if (response.status === 200) {
+      const contentType = parseContentType(response);
```
Comment (mozfreddyb) [inline app.js]: As above, pass resonse.headers.get("content-type") and deal with undefined/empty in the function itself.

File: app.js [hunk 4]
```diff
@@ -35,6 +35,44 @@ async function hashText(message) {
   return hash384;
 }
 
+function parseContentType(res) {
+  const type = res.headers.get("content-type").split(";");
```
Comment (mozfreddyb) [inline app.js]: What if the passed type is undefined? Please add an early-return for error cases that returns "script"

**Review by mozfreddyb — CHANGES_REQUESTED** [review]
Looks really solid, thank you!
Please remove .DS_Store from this commit (you may also add it to .gitignore, if you wish).
But let's finish reviewing this after #453 is done.

**Comment by kota-yata** [comment]
Ready to land :)

**Pull request merged**